src/include/btree_cmp_inline.h

```diff
@@ -21,6 +21,152 @@
+    tendp = tstartp + len;
+
+    /* skip 16 matching bytes at a time, starting at first possible difference. */
+    for (userp = ustartp, treep = tstartp; uendp - userp > WT_VECTOR_SIZE;
+         userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
+        memcpy(&udata, userp, WT_VECTOR_SIZE);
+        memcpy(&tdata, treep, WT_VECTOR_SIZE);
+        if (udata.a != tdata.a || udata.b != tdata.b)
+            goto final128;
+    }
+
+    /*
+     * Rewind until there is exactly 16 bytes left. We know we started with at least 16, so we are
+     * still in bound.
+     */
+    memcpy(&udata, uendp - WT_VECTOR_SIZE, WT_VECTOR_SIZE);
+    memcpy(&tdata, tendp - WT_VECTOR_SIZE, WT_VECTOR_SIZE);
+
+final128:
+    firsteq = udata.a == tdata.a;
+    u64 = firsteq ? udata.b : udata.a;
+    t64 = firsteq ? tdata.b : tdata.a;
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    u64 = __builtin_bswap64(u64);
+    t64 = __builtin_bswap64(t64);
+#endif
+
+    return (u64 < t64 ? -1 : u64 > t64 ? 1 : lencmp);
+}
+
+/*
+ * __lex_compare_lt_16 --
+ *     Lexicographic comparison routine for data less than 16 bytes. Returns: < 0 if user_item is
+ *     lexicographically < tree_item, = 0 if user_item is lexicographically = tree_item, > 0 if
+ *     user_item is lexicographically > tree_item. We use the names "user" and "tree" so it's clear
+ *     in the btree code which the application is looking at when we call its comparison function.
+ */
+static WT_INLINE int
+__lex_compare_lt_16(const uint8_t *ustartp, const uint8_t *tstartp, size_t len, int lencmp)
+{
+    uint64_t ta, tb, ua, ub, u64, t64;
+    const uint8_t *tendp, *uendp;
+
+    uendp = ustartp + len;
+    tendp = tstartp + len;
+    if (len & sizeof(uint64_t)) {
+        memcpy(&ua, ustartp, sizeof(uint64_t));
+        memcpy(&ta, tstartp, sizeof(uint64_t));
+        memcpy(&ub, uendp - sizeof(uint64_t), sizeof(uint64_t));
+        memcpy(&tb, tendp - sizeof(uint64_t), sizeof(uint64_t));
+    } else if (len & sizeof(uint32_t)) {
+        uint32_t ta32, tb32, ua32, ub32;
+        memcpy(&ua32, ustartp, sizeof(uint32_t));
+        memcpy(&ta32, tstartp, sizeof(uint32_t));
+        memcpy(&ub32, uendp - sizeof(uint32_t), sizeof(uint32_t));
+        memcpy(&tb32, tendp - sizeof(uint32_t), sizeof(uint32_t));
+        ua = ua32;
+        ta = ta32;
+        ub = ub32;
+        tb = tb32;
+    } else if (len & sizeof(uint16_t)) {
+        uint16_t ta16, tb16, ua16, ub16;
+        memcpy(&ua16, ustartp, sizeof(uint16_t));
+        memcpy(&ta16, tstartp, sizeof(uint16_t));
+        memcpy(&ub16, uendp - sizeof(uint16_t), sizeof(uint16_t));
+        memcpy(&tb16, tendp - sizeof(uint16_t), sizeof(uint16_t));
+        ua = ua16;
+        ta = ta16;
+        ub = ub16;
+        tb = tb16;
+    } else if (len & sizeof(uint8_t)) {
+        uint8_t ta8, ua8;
+        memcpy(&ua8, ustartp, sizeof(uint8_t));
+        memcpy(&ta8, tstartp, sizeof(uint8_t));
+        return (ua8 < ta8 ? -1 : ua8 > ta8 ? 1 : lencmp);
```

Comment:
We don't need the little endian check here?

Reply:
Here we are comparing bytes so no need to flip a single byte.